packages/api-client/src/components/IconSelector.vue

```diff
@@ -0,0 +1,57 @@
+<script setup lang="ts">
+import { RadioGroup, RadioGroupLabel, RadioGroupOption } from '@headlessui/vue'
+import { ScalarPopover } from '@scalar/components'
+import { LibraryIcon, libraryIcons } from '@scalar/icon-library'
+import { computed } from 'vue'
+
+const props = defineProps<{
+  modelValue: string
+  placement?: AlignedPlacement
+}>()
+
+const emit = defineEmits<{
+  (e: 'update:modelValue', icon: string): void
+}>()
+
+type Side = 'top' | 'bottom'
+type Alignment = 'start' | 'end'
+type AlignedPlacement = `${Side}-${Alignment}`
```

Comment:
nice!

Reply:
I stole this from inside floating ui 😂 But it is pretty slick